app/src/main/java/org/mozilla/fenix/library/history/HistoryController.kt

```diff
@@ -95,7 +108,39 @@ class DefaultHistoryController(
     }
 
     override fun handleDeleteSome(items: Set<History>) {
-        deleteHistoryItems.invoke(items)
+        val pendingDeletionItems = items.map { it.toPendingDeletionHistory() }.toSet()
+        appStore.dispatch(AppAction.AddPendingDeletionSet(pendingDeletionItems))
+        deleteSnackbar.invoke(items, ::undo, ::delete)
+    }
+
+    private fun undo(items: Set<History>) {
+        val pendingDeletionItems = items.map { it.toPendingDeletionHistory() }.toSet()
+        appStore.dispatch(AppAction.UndoPendingDeletionSet(pendingDeletionItems))
+    }
+
+    private fun delete(items: Set<History>): suspend (context: Context) -> Unit {
+        return { context ->
+            CoroutineScope(Dispatchers.IO).launch {
+                store.dispatch(HistoryFragmentAction.EnterDeletionMode)
+                for (item in items) {
+                    GleanHistory.removed.record(NoExtras())
```

Comment:
Should this part of `GleanMetricsService` and use the `metrics` to report ?  And following the same patter as on 
 `metrics.track(Event.HistorySearchTermGroupTapped)`

Reply:
I think it was done during working on `[this](https://github.com/mozilla-mobile/fenix/issues/24498)` issue.
I just brought that code from the fragment into the controller.